app/modules/import-export/assets/js/pages/import/import-kit/import-kit.js

```diff
@@ -144,7 +151,31 @@ export default function ImportKit() {
 					onError={ () => eventTracking( 'kit-library/modal-open', null, 'load', errorType, 'error' ) }
 					onLearnMore={ () => eventTracking( 'kit-library/seek-more-info', null, 'click', null, 'error' ) }
 				/> }
+
+				{ dialog.isOpen &&
+					<Dialog
+						title={ __( 'Warning: JSON or ZIP files may be unsafe', 'elementor' ) }
+						text={ __( 'Uploading JSON or ZIP files from unknown sources can be harmful and put your site at risk. For maximum safety, upload only JSON or ZIP files from trusted sources.', 'elementor' ) }
+						approveButtonColor="link"
+						approveButtonText={ __( 'Continue', 'elementor' ) }
+						approveButtonOnClick={ dialog.approve }
+						dismissButtonText={ __( 'Cancel', 'elementor' ) }
+						dismissButtonOnClick={ dialog.dismiss }
+						onClose={ dialog.dismiss }
+					>
+						<label htmlFor="do-not-show-upload-json-warning-again" style={ { display: 'flex', alignItems: 'center', gap: '5px' } }>
+							<Checkbox
+								id="do-not-show-upload-json-warning-again"
+								type="checkbox"
+								value={ checkbox.isChecked }
+								onChange={ ( event ) => checkbox.setIsChecked( !! event.target.checked ) }
```

Comment:
ref?

Reply:
The hook should be generic, without the assumption how implements with HTML checkbox